### PR TITLE
document terrascope job options

### PR DESCRIPTION
**Pull request by jdries**
job options need to be documented, as shown by forum questions: https://discuss.eodc.eu/t/error-occuring-when-calculating-s2-mosaic-for-some-periods-others-work-fine/492/3

### CODE REVIEW

**Comment by m-mohr** [comment]
See related PR: https://github.com/Open-EO/openeo-api/pull/471

File: federation/index.md [hunk 1]
```diff
@@ -176,13 +176,61 @@ on the appropriate processing back-ends.
...
+        "executor-memoryOverhead": "3G",
+        "executor-cores": "2",
+        "task-cpus": "1",
+        "executor-request-cores": "400m",
+        "max-executors": "100",
+        "driver-memory": "8G",
+        "driver-memoryOverhead": "2G",
+        "driver-cores": "5"
+    }
+cube.execute_batch( job_options=job_options)
```
Comment (soxofaan) [inline federation/index.md:207]:
```suggestion
cube.execute_batch(job_options=job_options)
```

File: federation/index.md [hunk 2]
```diff
@@ -176,13 +176,61 @@ on the appropriate processing back-ends.
...
+### Validity signed URLs (Batch job results)
 
 Batch job results are accessible to the user via signed URLs stored in the result assets. Within the platform, 
 these URLs have a validity (expiry time) of 7 days. Within these 7 days, the results of a batch job can be accessed 
 by any person with the URL. Each time a user requests the results from the job endpoint (`GET /jobs/{job_id}/results`), 
 a freshly signed URL (valid for 7 days) is created for the result assets.
 
+### Customizing batch job resources on Terrascope
+
+Jobs running on the (Terrascope) cluster get assigned a default amount of cpu and memory resources. This
```
Comment (soxofaan) [inline federation/index.md:188]:
```suggestion
Jobs running on the (Terrascope) cluster get assigned a default amount of CPU and memory resources. This
```

File: federation/index.md [hunk 3]
```diff
@@ -176,13 +176,61 @@ on the appropriate processing back-ends.
...
+        "driver-cores": "5"
+    }
+cube.execute_batch( job_options=job_options)
+```
+
+This is a short overview of the various options:
+
+- executor-memory: memory assigned to your workers, for the JVM that executes most predefined processes 
+- executor-memoryOverhead: memory assigned on top of the JVM, for instance to run UDF's
+- executor-cores: number of cpu's per worker (executor). The number of parallel tasks is executor-cores/task-cpus
```
Comment (soxofaan) [inline federation/index.md:214]:
```suggestion
- executor-cores: number of CPUs per worker (executor). The number of parallel tasks is executor-cores/task-cpus
```

File: federation/index.md [hunk 4]
```diff
@@ -176,13 +176,61 @@ on the appropriate processing back-ends.
...
+- task-cpus: cpus assigned to a single task. UDF's using libraries like Tensorflow can benefit from further parallellization on the level of individual tasks.
```
Comment (soxofaan) [inline federation/index.md:215]:
```suggestion
- task-cpus: CPUs assigned to a single task. UDF's using libraries like Tensorflow can benefit from further parallellization on the level of individual tasks.
```

File: federation/index.md [hunk 5]
```diff
@@ -176,13 +176,61 @@ on the appropriate processing back-ends.
...
+- executor-request-cores: this settings is only relevant for Kubernetes based backends, allows to overcommit cpu
```
Comment (soxofaan) [inline federation/index.md:216]:
```suggestion
- executor-request-cores: this settings is only relevant for Kubernetes based backends, allows to overcommit CPU
```

File: federation/index.md [hunk 6]
```diff
@@ -176,13 +176,61 @@ on the appropriate processing back-ends.
...
+- driver-memoryOverhead: memory assigned to the spark 'driver' on top of JVM memory, for Python processes.
+
+#### Learning more
+
+The topic of resource optimization is a complex one, and here we just give a short summary. The goal of openEO is to hide most of these
+details from the user, but we realize that advanced users sometimes want to have a bit more insight, so in the spirit of being open, we give some hints. 
+
+To learn more about these options, we point to the piece of code that handles this:
+
+https://github.com/Open-EO/openeo-geopyspark-driver/blob/master/openeogeotrellis/backend.py#L1213
```
[the remark this answers is not in the document]
Comment (jdries) [reply federation/index.md]: done

**Pull request merged**